sdk/helper/logging/logging.go

```diff
@@ -0,0 +1,81 @@
+package logging
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	log "github.com/hashicorp/go-hclog"
+)
+
+type LogFormat int
+
+const (
+	UnspecifiedFormat LogFormat = iota
+	StandardFormat
+	JSONFormat
+)
+
+// Stringer implementation
+func (l LogFormat) String() string {
+	switch l {
+	case UnspecifiedFormat:
+		return "unspecified"
+	case StandardFormat:
+		return "standard"
+	case JSONFormat:
+		return "json"
+	}
+
+	// unreachable
+	return "unknown"
+}
+
+// NewVaultLogger creates a new logger with the specified level and a Vault
+// formatter
+func NewVaultLogger(level log.Level) log.Logger {
+	return NewVaultLoggerWithWriter(log.DefaultOutput, level)
+}
+
+// NewVaultLoggerWithWriter creates a new logger with the specified level and
+// writer and a Vault formatter
+func NewVaultLoggerWithWriter(w io.Writer, level log.Level) log.Logger {
+	opts := &log.LoggerOptions{
+		Level:      level,
+		Output:     w,
+		JSONFormat: ParseEnvLogFormat() == JSONFormat,
+	}
+	return log.New(opts)
+}
+
+// ParseLogFormat parses the log format from the provided string.
+func ParseLogFormat(format string) (LogFormat, error) {
+
+	switch strings.ToLower(strings.TrimSpace(format)) {
```

Comment:
I like that you made these case insensitive, since I could see people providing "json" or "JSON", for instance.

Reply:
:-) I copied it over from how `ParseEnvLogFormat()` worked